workflows/dependabot.yml

```diff
@@ -0,0 +1,46 @@
+name: dependabot
+
+on:
+  pull_request:
+    branches:
+      - dependabot/**
+  push:
+    branches:
+      - dependabot/**
+  workflow_dispatch:
+
+permissions:
+  contents: write # Allow to update the PR.
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set up Go 1.x
+      uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f # tag=v3.3.1
+      with:
+        go-version: '1.19'
+      id: go
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # tag=v3.1.0
+    - uses: actions/cache@9b0c1fce7a93df8e3bb8926b0d6e9d89e92f20a7 # tag=v3.0.11
+      name: Restore go cache
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Update all modules
+      run: make generate-modules
+    - name: Update generated code
+      run: make generate
+    - uses: EndBug/add-and-commit@61a88be553afe4206585b31aa72387c64295d08b # tag=v9.1.1
+      name: Commit changes
+      with:
+        author_name: dependabot[bot]
+        author_email: 49699333+dependabot[bot]@users.noreply.github.com
```

Comment:
I am curious to know if `author_email` was auto-generated? If yes, how?

Reply:
I could not find any correlation to `49699333+dependabot[bot]@users.noreply.github.com` to dependabot. 

Dependabot's github page publicizes its email id as [support@github.com](mailto:support@github.com)
![Screenshot 2022-12-09 at 2 21 14 PM](https://user-images.githubusercontent.com/25679594/206805090-a9e9c86a-8d6d-4419-aded-3116de34d0fa.png)

one of the upgrade commit in my fork of CAPZ has dependabot's email ID as `support@github.com` too. Refer https://github.com/nawazkh/cluster-api-provider-azure/pull/1/commits/b7f7af4bf7cefa40c1527f2f281482cd955fe4a5

As this is just meta data for us repo watchers, I am ok with either `author_email: 49699333+dependabot[bot]@users.noreply.github.com` or `support@github.com`.

Reply:
Feel free to collapse this comment thread as resolved!